driver-core/src/main/com/mongodb/internal/operation/BulkWriteBatch.java

```diff
@@ -291,6 +291,9 @@ private BulkWriteResult getBulkWriteResult(final BsonDocument result) {
             if (!writeErrors.isEmpty()) {
                 writeRequests = writeRequests.filter(wr -> !writeErrors.contains(wr.getIndex()));
             }
+            if (payload.getPosition() < payload.size()) {
+                writeRequests = writeRequests.filter(wr -> wr.getIndex() <= payload.getPosition());
```

Comment:
I noticed we weren't removing any unprocessed inserts.

Reply:
Is there a test for this?  I can't tell if the one in the combiner spec is, but doesn't look like it.

Reply:
Good call - added and fixed the bug in this line!